### DISCUSSION TITLE
Welcome to Clash Verge Discussions!

### DISCUSSION THREAD

**Discussion by zzzgydi** [zzzgydi/clash-verge#29] [Announcements]
## 👋 欢迎讨论

  你可以：

  * 提出你的问题或遇到的bug
  * 分享你的想法
  * 提出改进的建议

**Comment by d2184** [comment]
目前缺个导入本地文件，得在应用内新建个本地的配置再复制粘贴。另外windows上在配置里选file打开配置文件的时候，会弹出failed to open file by 'open'，而实际上已经打开了

**Reply by zzzgydi** [reply]
> 另外windows上在配置里选file打开配置文件的时候，会弹出failed to open file by 'open'，而实际上已经打开了

这个是哪个版本，可以发一下log看看

**Reply by zzzgydi** [reply]
> 目前缺个导入本地文件，得在应用内新建个本地的配置再复制粘贴。

这个怎么实现好一点呢，可以将本地文件拖到窗口就完成导入，这样吗

**Reply by d2184** [reply]
> > 目前缺个导入本地文件，得在应用内新建个本地的配置再复制粘贴。
> 
> 这个怎么实现好一点呢，可以将本地文件拖到窗口就完成导入，这样吗

CFW上有专门的导入入口，拖动到窗口导入也是可行的

**Reply by d2184** [reply]
> > 另外windows上在配置里选file打开配置文件的时候，会弹出failed to open file by 'open'，而实际上已经打开了
> 
> 这个是哪个版本，可以发一下log看看

v0.023_x64_en-US，debug日志最后几行：
2022-03-15 12:30:05 - failed to open path "C:\Users\22390\.config\clash-verge\profiles\lRbWKMEZMkJG.yaml" for (code 1)
2022-03-15 12:30:14 - failed to open path "C:\Users\22390\.config\clash-verge\logs" for (code 1)

**Reply by zzzgydi** [reply]
CFW的可以点击选择本地文件，那就按这样的来实现吧。

弹出错误那里，我调整一下。

**Reply by d2184** [reply]
> CFW的可以点击选择本地文件，那就按这样的来实现吧。
> 
> 弹出错误那里，我调整一下。

OK，辛苦大佬了

**Comment by ileeoyo** [comment]
已知bug：tun模式无法正确解析域名

**Comment by Agoines** [comment]
Connctions 页面想法
![桌面端 HD](https://user-images.githubusercontent.com/50230626/158723995-542e9f01-7774-471e-92a3-94c4e2d8ffb8.png)

大概画了想要有的功能，但是我在 Mac 没有办法打开 Connctions 页面，连接正常，就没尝试自己去修改这个界面的 UI
![屏幕截图](https://user-images.githubusercontent.com/50230626/158724087-51525724-085d-42fa-bc52-66f25694f16d.jpg)

**Reply by zzzgydi** [reply]
Nice! Connections页面是正常打开了的，没有东西是因为你没有连接经过clash，所以是空白而已，不过有东西也只是十分简单的一个域名文本而已，啥也没做。

**Reply by Agoines** [reply]
> Nice! Connections页面是正常打开了的，没有东西是因为你没有连接经过clash，所以是空白而已，不过有东西也只是十分简单的一个域名文本而已，啥也没做。

不太清楚，Windows 那边没有问题，在 Mac 同样操作（Clash for Windows 退出了，访问谷歌没问题， 但是这个页面就是看不到，可能又是自己没操作对，到时候再说吧）我先看看其他功能我能不能帮上忙🤣

**Reply by zzzgydi** [reply]
开了系统代理了吗

**Reply by Agoines** [reply]
似乎是的，我开了系统代理重启以后就炸了，如果关闭系统代理，然后重启客户端再打开系统代理就好了

**Reply by zzzgydi** [reply]
炸了是指的啥问题

**Reply by Agoines** [reply]
> 炸了是指的啥问题

就界面这样了，流量图显打开了也没有，界面空空如也，但是配置也没坏
<img width="754" alt="截屏2022-03-17 上午11 23 30" src="https://user-images.githubusercontent.com/50230626/158730554-3f0f88c8-e72d-4e77-8f24-63795323b4d8.png">
<img width="754" alt="截屏2022-03-17 上午11 23 37" src="https://user-images.githubusercontent.com/50230626/158730564-eaf251ef-f09d-4550-b300-ced4594125e4.png">

**Reply by zzzgydi** [reply]
还是clash core启动问题，macOS的对吧

**Reply by Agoines** [reply]
> 还是clash core启动问题，macOS的对吧

对的，macOS

**Reply by zzzgydi** [reply]
我思考思考怎么优雅得处理这些错误，毕竟出现得比较频繁

**Reply by zzzgydi** [reply]
可能的原因是端口冲突和clash配置有错，或者和tun的配置有关

**Reply by Agoines** [reply]
运行 `yarn dev` 以后也一直在打印下面的这串错误，（不论 cfw 是否关闭，都会打印如下问题）
`ailed to activate for error sending request for url (http://127.0.0.1:9090/configs): error trying to connect: tcp connect error: Connection refused (os error 61)`

**Reply by zzzgydi** [reply]
这个是clash启动失败，然后请求不成功，要分析是啥错的话，还需要前面的log信息

**Reply by Agoines** [reply]
这是在 `yarn dev` 以后可以看到

 <img width="931" alt="截屏2022-03-17 上午11 44 41" src="https://user-images.githubusercontent.com/50230626/158732804-3d1d91c7-7a63-460b-acea-02bba70b633b.png">

最后就是一直在打印
`failed to activate for `error sending request for url (http://127.0.0.1:9090/configs): error trying to connect: tcp connect error: Connection refused (os error 61)``

**Reply by zzzgydi** [reply]
我猜测可能的原因就是，你之前打开yarn dev了，然后kill了这个进程后，启动的clash进程没有正常关闭，导致你再启动的时候，就permission denied了。

可以打开任务管理器，搜索clash进程，看看是不是存在，是的kill掉，再yarn dev一下，应该就好了。

**Reply by Agoines** [reply]
大概清楚了，是直接没启动，任务管理器找到的都 kill 了，但是还这样。你发的 Releases 可以用，但是有上面那个没启动的 bug，自己 yarn dev，也类似没启动的 bug这，个等你到时候更新了，我拉下来再看看，目前太玄学了。还有系统代理是不是要提权，我记得当时用cfw 也需要提权

**Comment by zzzgydi** [comment]
系统代理应该没有权限问题，即使开不了系统代理，也不应该影响clash的启动，不知道是啥问题，有空我再看看能不能复现

**Reply by Agoines** [reply]
> 系统代理应该没有权限问题，即使开不了系统代理，也不应该影响clash的启动，不知道是啥问题，有空我再看看能不能复现

我把问题就总结下，上面说的太乱了，

Releases
我开了系统代理重启客户端以后就 clash core 就没办法启动，如果关闭系统代理，然后重启客户端再打开系统代理就好了

yarn dev
不论开不开都是 clash core 都无法启动，就是 yarn dev 以后会出现上面的 log

**Reply by Agoines** [reply]
> 系统代理应该没有权限问题，即使开不了系统代理，也不应该影响clash的启动，不知道是啥问题，有空我再看看能不能复现

Releases 的问题清楚了
我的习惯是在下方的程序坞退出，结果有些进程没被杀死，但是顶部状态栏的图标没了，所以重启就出现 Clash Core 没载入的问题了。

另一个还不不清楚

**Reply by zzzgydi** [reply]
了解了解

**Reply by zzzgydi** [reply]
你本地yarn dev一直无法启动clash core吗，有可能是没有给clash那个文件一个可执行的权限。chmod 755 clash-xxxx

**Reply by Agoines** [reply]
> 你本地yarn dev一直无法启动clash core吗，有可能是没有给clash那个文件一个可执行的权限。chmod 755 clash-xxxx

Vs Code/clash-verge/src-tauri/sidecar/ 里面的那个clash？我全给读写了，说到这个我还想起来，在 window 这个 clash core 还下载不下来，就下载下来 tz，在别的目录，在 sidecar/ 没文件，当时自己下载丢进去了 exe 解决了，我怀疑是不是 `yarn run check` 有问题

**Reply by zzzgydi** [reply]
给读写还不够，要执行权限。下载不下来应该是墙的问题

**Reply by Agoines** [reply]
> 给读写还不够，要执行权限。下载不下来应该是墙的问题

我昨天不是

> 给读写还不够，要执行权限。下载不下来应该是墙的问题

chmod 755 clash-xxxx 以后，dev 还是看不到

**Reply by Agoines** [reply]
> 给读写还不够，要执行权限。下载不下来应该是墙的问题

window 的话似乎是不解压还是怎么回事，我当时替换成国内镜像提示下载成功了

**Reply by zzzgydi** [reply]
镜像链接发我看看

**Reply by zzzgydi** [reply]
macOS的话，`yarn run check --force`试试，看看有啥错没

**Reply by zzzgydi** [reply]
windows，你能终端走代理的话，也可以执行一遍`yarn run check --force`

**Reply by Agoines** [reply]
> 镜像链接发我看看

把 https://github.com/ 替换成 https://hub.fastgit.xyz/ 就好了

**Reply by Agoines** [reply]
> macOS的话，`yarn run check --force`试试，看看有啥错没

我去试试

**Reply by Agoines** [reply]
> windows，你能终端走代理的话，也可以执行一遍`yarn run check --force`
`yarn run check --force`
```
[INFO]: downloading from "https://hub.fastgit.xyz/Dreamacro/maxmind-geoip/releases/latest/download/Country.mmdb"
node:events:505
      throw er; // Unhandled 'error' event
      ^

Error: incorrect header check
    at Zlib.zlibOnError [as onerror] (node:zlib:189:17)
Emitted 'error' event on Gunzip instance at:
    at Gunzip.onerror (node:internal/streams/readable:796:14)
    at Gunzip.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:164:8)
    at emitErrorCloseNT (node:internal/streams/destroy:129:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -3,
  code: 'Z_DATA_ERROR'
}
Node.js v17.7.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

直接` yarn run check`
```
$ node scripts/check.mjs
✨  Done in 0.34s.

```

**Reply by zzzgydi** [reply]
换了这个域，测试了一下脚本也没问题的，不清楚哪里的问题了。

**Reply by Agoines** [reply]
> 换了这个域，测试了一下脚本也没问题，不清楚哪里的问题了。

那就不清楚，读写之类的权限也给了

**Reply by zzzgydi** [reply]
这个脚本也就是将一些文件下载到对应的目录而已，手动下载到对应的目录也行，只有clash执行文件是必须的，其他都是非必须的。总的来说问题不大。

**Reply by zzzgydi** [reply]
我想问问，还有类似 https://hub.fastgit.xyz/ 的稳定的gituhb资源的代理吗，还是说这个就比较稳定了呢。打算用来做软件更新。

**Reply by Agoines** [reply]
> 我想问问，还有类似 https://hub.fastgit.xyz/ 的稳定的gituhb资源的代理吗，还是说这个就比较稳定了呢。打算用来做软件更新。

我是当时发现这个仓库看到的，你可以看看
https://github.com/RC1844/FastGithub

**Reply by zzzgydi** [reply]
Ok, 用这个加速一下更新下载

**Reply by zzzgydi** [reply]
macOS 里 `yarn run check` 的问题我也遇到了，处理了一下，现在应该ok了

**Reply by Agoines** [reply]
> macOS 里 `yarn run check` 的问题我也遇到了，处理了一下，现在应该ok了

好哎，这个问题太玄学了

**Reply by zzzgydi** [reply]
这个应该是在下载clash的时候，链接不存在的问题

**Comment by Agoines** [comment]
实际上各家的节点分流不是很好，有时候有些比较小众的链接会被判断为类似这家的其他流量
<img width="535" alt="截屏2022-03-20 下午5 17 44" src="https://user-images.githubusercontent.com/50230626/159155849-9fed09b2-63fb-45eb-a63a-890d7d6c461e.png">
自己修改配置文件以后，再更新配置文件就会被覆盖，可不可以加一个类似这样的功能，在别人提供的后面追加自己的规则
<img width="784" alt="截屏2022-03-20 下午5 21 07" src="https://user-images.githubusercontent.com/50230626/159155963-079cb37d-c355-4f18-8b7a-570aabef1356.png">

**Reply by d2184** [reply]
是类似CFW这种配置文件预处理吧

parsers:
  - url: https://example.com/profile.yaml
    yaml:
      prepend-rules:
        - DOMAIN,test.com,DIRECT # rules最前面增加一个规则
      append-proxies:
        - name: test # proxies最后面增加一个服务
          type: http
          server: 123.123.123.123
          port: 456

**Reply by Agoines** [reply]
> 是类似CFW那种配置文件预处理吧

对的，但是我想法更奇怪一点，就是可以自己去开启和关闭这个规则

**Reply by zzzgydi** [reply]
这个完全支持的哈，可以看pinned issue的使用指南，关于profile有增强模式的

**Comment by Agoines** [comment]
关于更新弹窗
<img width="453" alt="截屏2022-03-20 下午5 22 24" src="https://user-images.githubusercontent.com/50230626/159155998-7d6639ec-230b-4f2e-b5b1-ffef0b5ff4d1.png">

可不可以在本地进行 Markdown 语法的解析并且渲染，应该会比会这个更直观点

**Reply by zzzgydi** [reply]
0.0.23版本是没做md解析的，但是0.0.24是做了的，所以这个版本之后的更新就没问题

**Reply by Agoines** [reply]
> 0.0.23版本是没做md解析的，但是0.0.24是做了的，所以这个版本之后的更新就没问题

okk，我还没更新，我等会更新看看

**Comment by xx183400** [comment]
![2bcf3da8fabebb4201dd38d743e256a](https://user-images.githubusercontent.com/103121833/161982218-3c378fd5-e1ca-4346-b302-a5a50c32b84f.png)
这个登陆不上 什么原因

**Reply by zzzgydi** [reply]
你确定host, 端口， 还有密钥这些都填对了吗。一般默认的控制端口是9090不是7890哦，7890是代理端口

**Comment by d2184** [comment]
最近用tun模式的次数多了，每次开启和退出都要进设置里操作有点麻烦，如果在系统托盘右键再加个tun模式的开关会更方便些

**Reply by zzzgydi** [reply]
可以

**Comment by DennisZY** [comment]
关于订阅，有没有可能做一个本地订阅转换呢？
https://github.com/Loyalsoldier/clash-rules 和 https://github.com/ACL4SSR/ACL4SSR 的规则的替换

**Reply by zzzgydi** [reply]
订阅转换的话，我对这个还不是很了解，是指类似说将ssr的订阅转成clash这种功能吗

**Reply by DennisZY** [reply]
是，但不止于ssr的订阅转成clash，其实我想用规则 https://github.com/Loyalsoldier/clash-rules 替换别人提供的规则。我不清楚Script 脚本处理能不能做到，主要是不会写hhh。

**Reply by ileeoyo** [reply]
> 是，但不止于ssr的订阅转成clash，其实我想用规则 https://github.com/Loyalsoldier/clash-rules 替换别人提供的规则。我不清楚Script 脚本处理能不能做到，主要是不会写hhh。

看看subconverter这个项目，搭建一个或者用别人提供的。可以各平台通用，单独clash-verge增加这个功能也没法再手机上用。

**Reply by zzzgydi** [reply]
@DennisZY 你只是想用这个项目的规则的话，可以新建一个`merge` profiles，然后将clash-rules项目里的rule-providers的配置拷贝到刚刚新建的merge profiles里，然后启用就完事了。

**Reply by DennisZY** [reply]
okk，好像可以了，感谢！

**Comment by mo-han** [comment]
配置中只有若干proxy-provider，但是并没有proxies。我想把这些provider中的节点聚合成一个proxies列表，这能用script配置实现吗？还是说，这个想法，从config层面压根就没法实现？

**Reply by mo-han** [reply]
我自己想办法实现了
